sdk/resource/resource.go

```diff
@@ -177,7 +205,9 @@ func Merge(a, b *Resource) (*Resource, error) {
 	case a.schemaURL == b.schemaURL:
 		schemaURL = a.schemaURL
 	default:
-		return Empty(), errMergeConflictSchemaURL
+		// Return the merged resource with an appropriate error. It is up to
+		// the user to decide if the returned resource can be used or not.
+		err = ErrSchemaURLConflict
```

Comment:
In this case, we return a resource without a URL.  Should we put something here so it does not cleanly merge a second time?

Reply:
I think returning an empty is the best we can do. It is accompanied with an error to signal to the user the error state and is documented that the resource will have no schema URL.

It may be the case that the user wants to set the schema URL which they would be able to do in the current setup by merging with a resource with an appropriate schema URL. They would not be able to do that if we restricted it.

Reply:
Actually, I think, `detect()` can now return a resource if the schema URL returned by the last detector or empty.

E.g. if we have:

```
resource1 with schema URL = 1
resource2 with schema URL = 2
resource3 with schema URL = 3
```

running detect (which makes merge in a loop) will return a merged resouree with schema URL = 3

E.g. if we have:

```
resource1 with schema URL = 1
resource2 with schema URL = 2
resource3 with schema URL = 3
```

running detect  will return a merged resource with schema URL = <empty>.

Am I correct?

Reply:
I don't follow @pellared what is that in reference to?

Reply:
This is the `detect` which I am referring to:

https://github.com/open-telemetry/opentelemetry-go/blob/242d23a1815d00f228e6767e4baea9019fefe50a/sdk/resource/auto.go#L51-L82

Reply:
I'm still not following.

Is you problem that in one situation detect will return a schema URL and if you take away a detector it wont? I don't follow how that is a problem.

Reply:
Won't detect also return an error if there is a merge conflict?

Reply:
Can you provide a clarification of what each schema URL is in your example and what you want to see?

Reply:
@MrAlias I edited my comment. I made many language and formatting mistakes 😉

Reply:
I think I see. So the issue is that for even number of detectors it will return a resource with no schema and for odd it will return the last?

What is proposed is that we should return a resource with no schema in all cases, right?

Reply:
> Won't detect also return an error if there is a merge conflict?

It will. But the returned resource will have a misleading Scheme URL that will possibly be later passed to tracer or meter provider.

If the user creates a resource with `New` and it will pass e.g. 3 detectors which will return (all) different scheme ULR. Then `New` will return a merged resource with scheme URL (from the resource returned by last detector) and a scheme conflict error.

Reply:
> What is proposed is that we should return a resource with no schema in all cases, right?

Correct. I think we should add a test case for it to avoid regression bugs.

Reply:
Updated.

Reply:
@MadVikingGod this tread was taken in a different direction than your original comment. Are you okay with my initial response? Can we resolve this?